ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/SchemaInfo.java

```diff
@@ -105,6 +128,21 @@ public int hashCode() {
                   .stream()
                   .map(f -> f.getName() + " " + f.getSchema().toTypeString())
                   .collect(Collectors.joining(", ", SqlBaseType.STRUCT + "<", ">")))
+          .put(
+              SqlBaseType.DECIMAL,
+              si -> {
+                final Object precision = si.getFieldProperties().get("precision");
+                final Object scale = si.getFieldProperties().get("scale");
+                String parameterString = "";
```

Comment:
consider throwing an exception if exactly one of precision/scale is present (that should be a bug) - and add a test for the behavior as well

Reply:
Apparently all constructors require both!